### DISCUSSION TITLE
Docker crashes, uses 100% of the CPU for hours, and returns 400 when visiting.

### DISCUSSION THREAD

**Discussion by nitlix** [Dokploy/dokploy#809] [Q&A]
Hey everyone, currently struggling with the stressful behaviour described above when using Dokploy. Using Ubuntu 24.0.4.1 LTS

I come back to the app crashed, a 400 page and 100% CPU usage. 
When restarting the instance, the page doesn't even show up at all. 

Logs gave this:
root@localhost:/# docker service logs dokploy
dokploy.1.qquthovjuti1@localhost    | 
dokploy.1.qquthovjuti1@localhost    | > dokploy@v0.11.1 start /app
dokploy.1.qquthovjuti1@localhost    | > node -r dotenv/config dist/server.mjs
dokploy.1.qquthovjuti1@localhost    | 
dokploy.1.qquthovjuti1@localhost    | Error: getaddrinfo ENOTFOUND dokploy-postgres
dokploy.1.qquthovjuti1@localhost    |     at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26) {
dokploy.1.qquthovjuti1@localhost    |   errno: -3008,
dokploy.1.qquthovjuti1@localhost    |   code: 'ENOTFOUND',
dokploy.1.qquthovjuti1@localhost    |   syscall: 'getaddrinfo',
dokploy.1.qquthovjuti1@localhost    |   hostname: 'dokploy-postgres'
dokploy.1.qquthovjuti1@localhost    | }
dokploy.1.qquthovjuti1@localhost    | Default middlewares already exists
dokploy.1.qquthovjuti1@localhost    | Network is already initilized
dokploy.1.qquthovjuti1@localhost    | Main config already exists
dokploy.1.qquthovjuti1@localhost    | Default traefik config already exists
dokploy.1.qquthovjuti1@localhost    | Error: getaddrinfo ENOTFOUND dokploy-postgres
dokploy.1.qquthovjuti1@localhost    |     at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26) {
dokploy.1.qquthovjuti1@localhost    |   errno: -3008,
dokploy.1.qquthovjuti1@localhost    |   code: 'ENOTFOUND',
dokploy.1.qquthovjuti1@localhost    |   syscall: 'getaddrinfo',
dokploy.1.qquthovjuti1@localhost    |   hostname: 'dokploy-postgres'
dokploy.1.qquthovjuti1@localhost    | }
dokploy.1.qquthovjuti1@localhost    | Postgres Started ✅
dokploy.1.qquthovjuti1@localhost    | Traefik Started ✅
dokploy.1.qquthovjuti1@localhost    | Error: getaddrinfo ENOTFOUND dokploy-postgres
dokploy.1.qquthovjuti1@localhost    |     at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26) {
dokploy.1.qquthovjuti1@localhost    |   errno: -3008,
dokploy.1.qquthovjuti1@localhost    |   code: 'ENOTFOUND',
dokploy.1.qquthovjuti1@localhost    |   syscall: 'getaddrinfo',
dokploy.1.qquthovjuti1@localhost    |   hostname: 'dokploy-postgres'
dokploy.1.qquthovjuti1@localhost    | }
dokploy.1.qquthovjuti1@localhost    | Redis Started ✅
dokploy.1.qquthovjuti1@localhost    | Setting up cron jobs....
dokploy.1.qquthovjuti1@localhost    | Migration failed Error: connect ECONNREFUSED 10.0.1.5:5432
dokploy.1.qquthovjuti1@localhost    |     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16) {
dokploy.1.qquthovjuti1@localhost    |   errno: -111,
dokploy.1.qquthovjuti1@localhost    |   code: 'ECONNREFUSED',
dokploy.1.qquthovjuti1@localhost    |   syscall: 'connect',
dokploy.1.qquthovjuti1@localhost    |   address: '10.0.1.5',
dokploy.1.qquthovjuti1@localhost    |   port: 5432
dokploy.1.qquthovjuti1@localhost    | }
dokploy.1.qquthovjuti1@localhost    | Main Server Error Error: connect ECONNREFUSED 10.0.1.5:5432
dokploy.1.qquthovjuti1@localhost    |     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16) {
dokploy.1.qquthovjuti1@localhost    |   errno: -111,
dokploy.1.qquthovjuti1@localhost    |   code: 'ECONNREFUSED',
dokploy.1.qquthovjuti1@localhost    |   syscall: 'connect',
dokploy.1.qquthovjuti1@localhost    |   address: '10.0.1.5',
dokploy.1.qquthovjuti1@localhost    |   port: 5432
dokploy.1.qquthovjuti1@localhost    | }

How exactly can this be fixed?

**Comment by Siumauricio** [comment]
When that happens you can do the following

docker service rm dokploy-postgres dokploy-redis dokploy

and then use this command

```dockerfile
  docker service create \
      --name dokploy \
      --replicas 1 \
      --network dokploy-network \
      --mount type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock \
      --mount type=bind,source=/etc/dokploy,target=/etc/dokploy \
      --mount type=volume,source=dokploy-docker-config,target=/root/.docker \
      --publish published=3000,target=3000,mode=host \
      --update-parallelism 1 \
      --update-order stop-first \
      --constraint 'node.role == manager' 
      dokploy/dokploy:latest
```
